%%%%%%
%%
%%  Don't reorder the reviewer points; that'll mess up the automatic referencing!
%%
%%%%%

\begin{minipage}[b]{2.5in}
  Revision Cover Letter \\
  {\it Evolution}
\end{minipage}
\hfill
\begin{minipage}[b]{2.5in}
    Joshua Schiffman \\
    \emph{and} Peter Ralph \\
  \today
\end{minipage}
 
\vskip 2em
 
\noindent
{\bf To the Editor(s) -- }
 
\vskip 1em

 Cover letter. 


\noindent \hspace{4em}
\begin{minipage}{3in}
\noindent
{\bf Sincerely,}

\vskip 2em

{\bf 
Joshua Schiffman and
Peter Ralph
}\\
\end{minipage}

\vskip 4em

\pagebreak

%%%%%%%%%%%%%%
\reviewersection{AE}
\begin{quote}
This is a very interesting theory paper that connects the modeling and evolution of regulatory networks with speciation. While quite theoretical in nature, the authors made an effort to provide examples, and I consider this paper of very high relevance for the field.

The authors have addressed most of the previous comments from the first round of reviews, and the two new reviewers only raise minor concerns and suggestions, especially regarding the paragraph on Haldane’s rule and regarding a better placement of the paper in the context of speciation research. I agree with these concerns; regarding the latter point I have three additional questions/comments, one of which was also raised in the previous round of reviews:
\end{quote}

Thank you for the kind words. 

\begin{point}{Gene flow}
As it is described, the model only holds for allopatric populations; I would expect even a small amount of gene flow to make neutral evolution of hybrid incompatibility impossible (Gavrilets 1997, Bank et al. 2012). This is not discussed in the current version of the paper; even more interestingly, would the authors expect conditions in which emergence of incompatibility could be possible in this model also in the presence of gene flow?
\end{point}

\reply{
This is correct. As the model is currently described, we show that since different mutationally connected systems are not usually reproductively compatible, populations in different parts of system space will produce unfit hybrids. We rely on results from quantitative genetics to estimate the rate at which systems drift through the neutral space and thus to estimate the effects of gene flow. Without further modeling we do not yet know how a quantitative genetics model incorporating system theory would differ, if it all, from these classic results. We are planning to submit a paper where we further develop the removed quantitative genetics model and may be able to address this interesting question there.

We also briefly mention that gene flow will likely complicate matters in the discussion \revref.
}

\begin{point}{Orr's Snowball}
The authors state that rapid emergence of incompatibility is possible in their model. How would the accumulation of incompatibility here compare with Orr’s ``snownball'' model, and how is this expected to depend on the complexity of the underlying networks?
\end{point}

\reply{\jss{I think Orr's estimate of the number of incompatible sites should hold? Maybe we should clarify that what we calculate is the rate at which phenotypes breakdown/fitness decays, and not the number of incompatible genes? I'm not exactly sure how to approach this question. This seems related to one of our previous discussions about whether or not this model produces ``DMIs'' or something rather different.}
}

\begin{point}{Genetic architecture}
The genetic architecture (e.g., interactions between cis-regulatory and coding regions) is important to determine the barrier to gene flow that an incompatibiltiy builds. If there is a lot of short-distance interactions (physically in the genome; i.e. low recombination), they might not play a role for speciation per se because the incompatibility will often not be exposed to selection even in hybrids. In the light of our current knowledge of the genetic archtiecture of regulatory networks, how would this affect the conclusions from the model?
\end{point}

\reply{This is an interesting question. In our model system coefficients are treated as independent heritable units, and thus would represent genes spread out sufficiently far apart along the genome. Since the major source of F2 hybrid breakdown in this model is from recombining regulatory coefficients, if the genetic architecture of these coefficients was such that the genes were close enough to not recombine, we would not see the stark pattern of hybrid breakdown that we observe in F2 oscillators. We suspect that genetic architectures that allow for recombination, albeit at a lower rate, than for architectures in which genes are very far apart, will diminish the speed of speciation. This also leads to interesting questions about the evolution of genetic architecture; perhaps arhitectures in which co-regulating genes are neighbors are more robust.

We briefly mention complications to the model in the discussion \revref.
}

\reviewersection{1}
\begin{quote}
  I am basically enthusiastic about this paper, which has been influential for some time on bioRxiv. The central sections, especially e.g. \hyperlink{rev1:1}{lines 127-219}, are very clear and novel. I do think, though that the work could be better placed in context.
\end{quote}

Thank you. We were also excited that the paper had many readers on the bioRxiv. 

\begin{point}{Empirical evidence for the (minor?) role of drift in speciation:}
%1. Empirical evidence for the (minor?) role of drift in speciation:
%
  \hyperlink{rev1:2}{Lines 382-389} discuss previous discussion about the role of drift in speciation, but they focus on theory. I think that some strong arguments for the primacy of selection in speciation have been empirical (bottlenecking Drosophila populations and testing for RI, MK-tests on ``speciation genes'' etc.). Some of the earlier work was summarized in the last chapter of Coyne and Orr 2004. \hyperlink{rev1:3}{Line 396} seems to ignore a lot of this work, and so I would rephrase.
\end{point}

\reply{This is an important point. We have added references to the experimental speciation literature \revref. 
}

\begin{point}{Additive quantitative trait models}
%2. Additive quantitative trait models
%
The previous version of this manuscript contained a standard quantitative trait model. I think it was a good idea to remove this material, but I still think that this modelling approach (including Fisher’s geometric model) needs to be discussed a little more here. In the introduction, for example, it seems important to state clearly that ``kryptotypes'' (many-to-one genotype-to-phenotype maps) are also a feature of these standard models, and that previous authors have shown how phenotypically identical populations can evolve RI, in some cases via drift alone (e.g., Mani and Clarke 1990; Rosas et al. 2010; Fraisse et al. 2016 and others).

Martin 2014 argued that the results of these models and ``systems-biology'' type models are likely to approximate one another under some conditions, and that could be mentioned.
\end{point}

\reply{In the last paragraph of the introduction we discuss exactly this and the corresponding literature, including Rosas et al 2010 and Fraisse et al 2016. \jss{I don't really think Mani and Clarke says what R1 says it does. It states that mutational order matters. Also I don't thik the introduction is a good place for Martin 2014 -- maybe system and quant gen models have some sort of correspondence, but we don't show it.}
}

\begin{point}{RI due to kryptotypes not involving drift}
%3. RI due to kryptotypes not involving drift
%
  The section beginning on \hyperlink{rev1:4}{line 419} seems quite important to me, but it seems to downplay an important point. Populations might diverge genetically, or in their kryptotypes, but not in their phenotypes, if they experience fluctuating selection. This goes back the Wright-Fisher debates, and has been discussed in the context of speciation by, e.g., Mani and Clarke 1990, Barton 2001 and later authors. I think that the empirical evidence for fluctuating selection is also quite strong (e.g Bell 2010), and so it seems more important to discuss that, e.g., selection for robustness.
\end{point}

\reply{This is also a very interesting point worth discussing. We have added discussion on fluctuating selection \revref.
\jss{Not sure where to put Mani and Clarke yet}.
}

\begin{point}{Interpretation of the Haldane’s Rule result}
%4. Interpretation of the Haldane’s Rule result
%
  The abstract refers to ``another possible explanation'' of Haldane's Rule, and the explanation given is said to be quite different from the ``dominance theory'', because ``it derives from the nature of segregation variance''.  I wonder whether this is not misleading. My view is that the sort of model explored here (like additive quantitative trait model used by Barton 2001) naturally generates the sort of dominance relations required for the ``dominance theory'' to hold. Unless the authors strongly disagree, I would modify the abstract and \hyperlink{rev1:5}{lines 278-291}, or at least provide a fuller clarification for why the expanation is different from that discussed by Barton 2001.
\end{point}

\reply{
  The connection between the ``dominance theory'' and the situation presented here does not seem that straight forward to us. Haldane's rule should occur here even if the alleles are all strictly additive. Depending on the specific genetic architecture of a system, the dominance coefficient may not even be defined. For instance, if we consider a mutation that changes the $\tau$ parameter in the oscillator, $h$ as typically formulated in undefined. If we consider mutation to the particular coefficients themselves, $h$ varies sigmoidally from recessive to dominant, depending on the magnitude of the change (and the fitness function). Without an explicit model of mutation it is not clear whether these changes are dominant or not.

In this model we are also thinking about the effect of many small/quantitative mutations -- the effect of which on fitness, the cumulative effect of which could either be ``dominant'' or ``recessive''.

Figure 5 in Barton 2001 shows that F2 hybrid breakdown under \emph{additive} inheritence. Our model links F2 hybrid breakdown with heterogametic inviability -- in our model F1 heterogametes have lower fitness than homogametes because their system looks like an autosomal F2 system. 
}

\begin{point}{}
  (very minor) \hyperlink{rev1:6}{Line 358}: As you say, F2 hybrid breakdown has been known about for a very long time. As such, the single citation used feels a bit arbitrary.
\end{point}

\reply{Removed the citation. 
}

\reviewersection{2}
\begin{quote}
The manuscript explores how regulatory changes yielding the same phenotype may lead to hybrid incompatibilities and speciation. The authors model gene regulatory networks as sets of linear equations, where environmental inputs are integrated, giving rise to specific phenotypic outputs. Using Kalman decomposition, the authors identify the subset of such systems that produce the same phenotype and thus, characterise how a given system may be modified without affecting its phenotypic response. Resorting to a simple oscillatory network as an example, the authors show that phenotypically identical parents can generate offspring with considerably different phenotypes, resulting in hybrid incompatibilities if the set of equivalent phenotypes is not closed under the average. The phenotypic distance between parents and offspring is proportional to the square of the genetic distance between parents for F1 hybrids and directly proportional to the genetic distance between parents for F2 hybrids.

This is an interesting paper that nicely bridges core concepts of systems biology and evolution. The examples provided are very ``visual'', making quite clear the idea of ``equivalent systems'' and will greatly assist readers not familiar with Kalman decomposition. The figures are both informative and elegant, and the discussion aptly covers model validity and makes an effort to relate this work to standard speciation models. In addition, the authors have adequately addressed the comments of previous reviewers. Overall, this manuscript constitutes an appropriate addition to Evolution.

We recommend this paper for publication, but we have some suggestions that we consider would improve the clarity of this paper for a broader biological audience.
\end{quote}

Thank you. 

\begin{point}{Haldane's rule}
Our primary concern with the text regards Haldane's rule. The authors state that a male F1 offspring has a phenotype that could usually only be obtained in an F2 individual if all the involved genes were autosomal. Although this is true, it does not look like an arbitrary F2 individual but one with a very particular structure. This type of individual is like F1 in all autosomal genes but equal to the parental genotype in the genes present in the sex chromosome. It is not clear to us that this should necessarily generate a worse outcome than a female individual.

The F2 individuals only perform worse than F1 on average, and looking at figure 4 (left), we can see that some F2 individuals actually perform better than the F1 individuals.
\end{point}

\reply{
  The fitness of heterogametic F1s and F2s will always depend on the genetic distance between parents and on the particular gene network. Thus it is not guaranteed that the heterogametic F1s, will always have lower fitnesses than homogametic F1s. This is clear when the genetic distance is zero between parents. The model does show, however, that the particular combination of genes found in F1 heterogametes would only be possible in F2 autosomal crosses. Our model shows that phenotypes generated by networks contained on the autosomes have lower fitnesses in F2s than in F1s, and by extension, lower fitness in the heterogametic sex when networks are contained, at least in part, on the sex chromosomes.

  It is also true that some F2 individuals will perform as well as F1s and F0s -- for instance it is possible to recover an F0-like genotype (and therefore phenotype) after two generations of hybridization. In this model, hybrids have lower fitness because of increased genetic variance. However, there will also be F2 individuals with genes from both parental types -- meaning that there will be more low fitness F2s than low fitness F1s and F0s. \jss{(The F2 fitness is lower on average?)}
}

\begin{point}{}
  \hyperlink{rev2:1}{(p. 2, l. 73)} \jss{Note: R2 is using line numbers from the diff}: This paragraph summarises the main body of work performed in this paper. Thus, consider moving it to the end of the Introduction (i.e. switching places with the last paragraph).
\end{point}

\reply{Thank you for this suggestion. We've decided to keep the paragraph order the same though. 
}

\begin{point}{}
  \hyperlink{rev2:2}{(p. 2, l. 92)}: ``...more than one way to do the same thing...'' sounds a bit too colloquial. Consider changing it to ``...more than one way to accomplish the same outcome...'' or some other alternative phrasing.
\end{point}

\reply{Thank you for the suggestion, but we would prefer to keep the sentence as written. 
We understand your concern, but in order to make this paper easy to read,
we have avoided using acronyms and tried to use simple language wherever possible,
so long as the meaning remains clear.
}

\begin{point}{}
  \hyperlink{rev2:3}{(p. 5, l. 192)}: ``The Kalman decomposition then classifies each direction in kryptotype space as either reachable or unreachable''. Is there a biological motivation for considering unreachable directions in kryptotype space as degrees of freedom of the neutral space?
\end{point}

\reply{This is a good question. The unreachable subspace would correspond to system structures that do not have an effect on the phenotype. However, biologically, it could be understood as a form of cryptic genetic variation -- coefficients in the unreachable subspace could have an effect when a system is perturbed (e.g. mutationally or in a hybrid). \jss{Not sure if/where to add to the paper?}
}

\begin{point}{}
\hyperlink{rev2:4}{(p. 6, l. 208)}: ``both the change of basis used to obtain the decomposition and, once in this form, all submatrices other than $A_{ro}$, $B_{ro}$, and $C_{ro}$ can be changed without affecting the phenotype, and so represent degrees of freedom''. A minor question on generality - does Kalman decomposition require that the coefficient matrices are time-invariant?
\end{point}

\reply{
  Yes, the Kalman decomposition is for linear time-invariant systems. We now note this \revref{}. 
}

\begin{point}{Recombination}
(p. 7) In the section ``Sexual reproduction and recombination'', we think that the recombination process is not clearly explained, making it more challenging to read the text from this point. We suggest that the authors clarify how exactly recombination is implemented in the model. Perhaps an example would help the reader to have a better picture of the process. This example could be inline, like the one introduced in the section on Haldane's rule.
\end{point}

\reply{We do not think we can add an example of how recombination generally works -- the mechanism will depend on the details of the particular system under study. In the oscillator example, recombination independently draws system coefficients from each parent at random. 
}

\begin{point}{}
(p. 8, Figure 4 and p. 10, Figure 5): We suggest dividing the figure into sub-figures (e.g. A and B). Furthermore, we suggest adding ``F1'' and ``F2'' labels to the left and right plots in Figure 5, respectively, as well as the values of epsilon. This way, the reader would be immediately informed of what is changing.
\end{point}

\reply{\jss{To do.}
}

