\documentclass{article}
\usepackage{amsmath, amssymb, color, xcolor, amsthm}
\usepackage{graphicx, wrapfig, float, caption, dsfont, bbm, xfrac}
\usepackage{fullpage}
\usepackage[backref=page, hidelinks, colorlinks=true, citecolor=blue!60!black!100]{hyperref}
\usepackage{tikz}
\usetikzlibrary{arrows.meta, shapes}
\usepackage{caption, subcaption}
\usepackage{natbib} % gives us \citet: Author (year) and \citep: (Author; year)
\usepackage{authblk}
\usepackage{lineno}
\usepackage{multicol}
\usepackage[bottom]{footmisc}

% for comments in margins or not
\newif\ifmargincomments
\margincommentsfalse
% \margincommentstrue
\ifmargincomments
    \usepackage{todonotes}
    \usepackage[left=1cm,right=6.5cm,top=3cm,bottom=3cm,nohead,nofoot,marginparwidth=6cm]{geometry}
    \newcommand{\plr}[1]{\todo[color=blue!25]{#1}}
    \newcommand{\js}[1]{\todo[color=green!25]{#1}}
    % inline comment version
    \newcommand{\plri}[1]{{\color{blue}\it #1}}
\else
    \newcommand{\plr}[1]{{\color{blue}\it #1}}
    \newcommand{\js}[1]{{\color{green}\it #1}}
    \newcommand{\plri}[1]{\plr{#1}}
\fi

\newcommand{\jss}[1]{{\color{olive}\it #1}}
% \newcommand{\ddt}{\frac{d}{dt}}
\newcommand{\ddt}{\dot}
\newcommand{\ro}{{ro}}
\newcommand{\nro}{{\bar{r}o}}
\newcommand{\rno}{{r\bar{o}}}
\newcommand{\nrno}{{\bar{r}\bar{o}}}
\newcommand{\reachable}{\mathcal{R}}
\newcommand{\unobservable}{\bar{\mathcal{O}}}
\newcommand{\R}{\mathbb{R}}
\newcommand{\E}{\mathbb{E}}
\renewcommand{\P}{\mathbb{P}}
\newcommand{\var}{\mathop{\mbox{Var}}}
\newcommand{\cov}{\mathop{\mbox{Cov}}}
\newcommand{\tr}{\mathop{\mbox{tr}}} % trace
\newcommand{\pda}{\frac{\partial}{\partial A_{ij}}}
\newcommand{\ind}{\mathds{1}}
\newcommand{\grad}{\nabla}

\newcommand{\calH}{\mathcal{H}}
\newcommand{\diag}{\text{diag}}
\newcommand{\1}{\mathbbm{1}}

% the triple (A,B,C) as 'system' (not using)
\newcommand{\Sys}{\mathcal{S}}
% neutral set of all systems
\newcommand{\allS}{\mathcal{N}}

% fitness as a fn of distance
\newcommand{\fit}{\mathcal{F}}
% fitness as a fn of A
\newcommand{\fitx}{\mathcal{F}}
% set of optimal coefficients
\newcommand{\optx}{\mathcal{X}}
% optimal phenotype
\newcommand{\optph}{\Phi_0}
% distance in phenotype space
\newcommand{\dph}{d}
% incompatibility
\newcommand{\Incompat}{\mathcal{I}}

\DeclareMathOperator{\spn}{span}

\newtheorem{theorem}{Theorem}
\newtheorem{lemma}{Lemma}
\newtheorem{definition}{Definition}
\newtheorem{example}{Example}

\begin{document}
%\linenumbers

{\centering
{\Huge \bf System drift and speciation} \\ \vspace{0.75cm}
Joshua S. Schiffman$^{\dagger}$ \qquad Peter L. Ralph$^{\ddagger}$ \\ \vspace{0.5cm}
$^{\dagger}${\footnotesize {New York Genome Center, New York, New York 10013, U.S.A \& Weill Cornell Medicine, New York, New York 10065, USA  \\
$^{\ddagger}$Departments of Mathematics and Biology \& The Institute for Ecology and Evolution, University of Oregon, Eugene, Oregon 97403, USA}} \\ \vspace{0.5cm}
{\small \texttt{jschiffman@nygenome.org} \qquad 
\texttt{plr@uoregon.edu}} \\ \vspace{0.5cm}
%\small \today \\
\vspace{0.25cm}
}


\begin{abstract}
Even if a species' phenotype does not change over evolutionary time, 
the underlying mechanism may change, 
as distinct molecular pathways can realize identical phenotypes.
Here we use linear system theory to explore the consequences of this idea,
describing how a gene network underlying a conserved phenotype evolves,
as the genetic drift of small changes to these molecular pathways
cause a population to explore the set of mechanisms with identical phenotypes.
To do this, we model an organism's internal state as a linear system of differential equations
for which the environment provides input and the phenotype is the output,
in which context there exists 
an exact characterization of the set of all mechanisms that give the same input--output relationship.
This characterization implies that selectively neutral directions in genotype space should be common
and that the evolutionary exploration of these distinct but equivalent mechanisms
can lead to the reproductive incompatibility of independently evolving populations.
This evolutionary exploration, or \emph{system drift}, 
is expected to proceed at a rate proportional to the amount of intrapopulation genetic variation
divided by the effective population size ($N_e$).
At biologically reasonable parameter values
this could lead to substantial interpopulation incompatibility,
and thus speciation, on a time scale of $N_e$ generations.
This model also naturally predicts Haldane's rule, 
thus providing another possible explanation
of why heterogametic hybrids tend to be disrupted more often than homogametes 
during the early stages of speciation.
\end{abstract}

\textbf{Key words:} Speciation, Models/Simulations, Genetic Drift

\section*{Acknowledgements}


    We would like to thank Sergey Nuzhdin, Stevan Arnold, Michael Turelli, Patrick Phillips, Erik Lundgren and Hossein Asgharian for valuable discussion. 
    We would also like to thank Nick Barton, Sarah Signor, Todd Parsons, and Joachim Hermisson for very helpful comments on the manuscript.
    Work on this project was supported by funds from
    the Sloan Foundation and the NSF (under DBI-1262645) to PR.


\end{document}
